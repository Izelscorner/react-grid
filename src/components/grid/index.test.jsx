import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Grid from './';

describe('<Grid />', () => {
    let wrapper;
    let fetchPersonData = '';
    let columns = [{columnHeader: 'Name', columnField: 'name'},{columnHeader: 'Lastname', columnField: 'lastname'}]
    let resp = {data: {count: 2, results: [{name: 'test', lastname: 'something'}, {name: 'test', lastname: 'something'}]}};
    beforeAll(() => {
        fetchPersonData =jest.fn().mockImplementation(() => Promise.resolve(resp));
        wrapper = shallow(<Grid dataHandler={fetchPersonData}
            paginationParam={'page'}
            searchParam={'search'}
            columns={columns} />);
    });

    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Check Table Props', () => {
        const table = wrapper.find('Table');
        expect(table.props().data.length).toEqual(2);
        expect(table.props().columns.length).toEqual(2);
        expect(table.props().isLoading).toEqual(false);
    });

    it('Check Pagination Props', () => {
        const pagination = wrapper.find('Pagination');
        expect(pagination.props().count).toEqual(2);
        expect(pagination.props().currentPage).toEqual(1);
        expect(pagination.props().paginationParam).toEqual('page');
        expect(pagination.props().size).toEqual(2);
        expect(typeof pagination.props().paginationHandler).toEqual('function');
    });

    it('Check Search Props', () => {
        const pagination = wrapper.find('Search');
        expect(pagination.props().searchParam).toEqual('search');
        expect(typeof pagination.props().searchHandler).toEqual('function');
    });
});
