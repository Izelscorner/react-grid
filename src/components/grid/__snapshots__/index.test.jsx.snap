// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <Search
        searchHandler={[Function]}
        searchParam="search"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <Table
        columns={
          Array [
            Object {
              "columnField": "name",
              "columnHeader": "Name",
            },
            Object {
              "columnField": "lastname",
              "columnHeader": "Lastname",
            },
          ]
        }
        data={
          Array [
            Object {
              "lastname": "something",
              "name": "test",
            },
            Object {
              "lastname": "something",
              "name": "test",
            },
          ]
        }
        isLoading={false}
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col align-self-center"
    >
      <Pagination
        count={2}
        currentPage={1}
        paginationHandler={[Function]}
        paginationParam="page"
        size={2}
      />
    </div>
  </div>
</div>
`;

exports[`<Index /> matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <Search
        searchHandler={[Function]}
        searchParam="search"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <Table
        columns={
          Array [
            Object {
              "columnField": "name",
              "columnHeader": "Name",
            },
            Object {
              "columnField": "lastname",
              "columnHeader": "Lastname",
            },
          ]
        }
        data={
          Array [
            Object {
              "lastname": "something",
              "name": "test",
            },
            Object {
              "lastname": "something",
              "name": "test",
            },
          ]
        }
        isLoading={false}
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col align-self-center"
    >
      <Pagination
        count={2}
        currentPage={1}
        paginationHandler={[Function]}
        paginationParam="page"
        size={2}
      />
    </div>
  </div>
</div>
`;
